clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/secured/AccessTokenRetrieverFactory.java

```diff
@@ -70,6 +73,8 @@ public static AccessTokenRetriever create(Map<String, ?> configs,
             if (jou.shouldCreateSSLSocketFactory(tokenEndpointUrl))
                 sslSocketFactory = jou.createSSLSocketFactory();
 
+            boolean urlencodeHeader = validateUrlencodeHeader(cu);
```

Comment:
nit: `validateUrlEncodeHeader`

Reply:
I've seen the use of `urlencode` (all one token) as a fairly accepted way to join the words 'URL' and 'encode'. For example:

* [IANA](https://www.iana.org/assignments/media-types/application/x-www-form-urlencoded)
* [PHP](https://www.php.net/urlencode)
* [Python](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode)

In the JDK’s `URLEncoder` class, it references the MIME type `application/ x-www-form-urlencoded`.

In addition, the original code that introduced the encoding option uses the configuration `SASL_OAUTHBEARER_HEADER_URLENCODE`, which should arguably be `SASL_OAUTHBEARER_HEADER_URL_ENCODE`.

Personally, `urlencode` is a bit funny-looking to me, but I was attempting to maintain internal consistency with what appears to be an accepted way to capitalize that token.

Is it OK to leave it as is for now?